pandas/io/packers.py

```diff
@@ -390,7 +390,8 @@ def encode(obj):
 
             # store tz info and data as UTC
             if tz is not None:
-                tz = u(tz.zone)
+                tz = u(tz.zone) if not isinstance(tz, _FixedOffset) \
```

Comment:
What happens if the timezone is not a FixedOffset? e.g. `pytz.UTC` or `dateutil.tz.tzutc`?

Reply:
@mroeschke 
Encoding process of TimeStamp is using zone attribute and normalize the datetime to utc.
Now there is no zone attribute in FixedOffset hence we will not be able to encode it and loosing the offset value in the process.
So i extracted offset value and encode that.

Reply:
```python
>>> Timestamp('2018-12-02 14:50:00',tz='UTC').tzinfo.zone
'UTC'
>>> Timestamp('2018-12-02 14:50:00',tz=pytz.FixedOffset(-420)).tzinfo.zone is None
True

```

Reply:
Got it. Thanks for the explanation.

I'd change this to a `try`/`except` block instead of an `if`/`else`. 

Another option would be to `str` the timezone otherwise?

```
In [3]: str(tz)
Out[3]: 'pytz.FixedOffset(60)'

In [4]: import dateutil

In [5]: str(dateutil.tz.tzutc())
Out[5]: 'tzutc()'

In [7]: dateutil.tz.tzutc().zone
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-d8163524e913> in <module>()
----> 1 dateutil.tz.tzutc().zone

AttributeError: 'tzutc' object has no attribute 'zone'
```

Reply:
@mroeschke
Converting the timezone object into string to encode is a great idea.
But I will have to use something like this in decoding side.

```python
tz = eval('pytz.FixedOffset(60)' )
```
Using eval doesn't seem like a good idea.

Reply:
Agreed. I'd prefer your minutes extraction method then.

I imagine there will need to be separate handling for dateutil timezones all together.